core/src/main/java/com/linecorp/armeria/client/RedirectingClient.java

```diff
@@ -138,15 +139,19 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
             final HttpRequestDuplicator reqDuplicator = req.toDuplicator(ctx.eventLoop().withoutContext(), 0);
             execute0(ctx, redirectCtx, reqDuplicator, true);
         } else {
-            req.aggregateWithPooledObjects(ctx.eventLoop(), ctx.alloc()).handle((agg, cause) -> {
-                if (cause != null) {
-                    handleException(ctx, null, responseFuture, cause, true);
-                } else {
-                    final HttpRequestDuplicator reqDuplicator = new AggregatedHttpRequestDuplicator(agg);
-                    execute0(ctx, redirectCtx, reqDuplicator, true);
-                }
-                return null;
-            });
+            req.aggregate(AggregationOptions.builder()
+                                            .usePooledObjects(ctx.alloc())
+                                            .executor(ctx.eventLoop())
+                                            .build())
```

Comment:
This pattern seems frequently used. How about adding a factory method for it?

```suggestion
            req.aggregate(AggregationOptions.usePooledObjects(ctx.alloc(), 
                                                              ctx.eventLoop())
```